### DISCUSSION TITLE
Is there a way to see whether a callback has been called via the usage log?

### DISCUSSION THREAD

**Discussion by rilhia** [deepgram/community#100] [General help] [answered]
Hi all,

I have just started playing around and have stumbled upon an issue I cannot debug. I am using the API and have been able to make requests via the transcription endpoint synchronously (for relatively small files), but had to move to doing this asynchronously when I tried larger wav files. To use the callback functionality, I am using pipedream.com to provide me with an endpoint. First, I tested this with some small files and it worked perfectly. But when I tried files over 100MB, the processing appears to take place (according to the usage log), but there is no evidence of the callback url being called. I see no evidence of this in piperdream.com and cannot tell from the usage log in Deepgram whether the callback url has actually been called. This could very well be an issue with pipedream.com, but I cannot see whether the callback url has definitely been called and if so, what time it was called. Am I missing something here, or is this data not currently provided?

**Comment by jviso** [comment, marked as answer]
Hi @rilhia. You're right that Deepgram doesn't record the callback attempts in the usage logs within the Deepgram Console. That might be a great addition to the product!

Here are two thoughts to consider for your debugging:

1. Try using a callback server on your local machine. One quick way to do this would be to use `nc` and `ngrok` in separate terminal sessions, e.g. `nc -l -p 5000` in one session and `ngrok http 5000` in another; then, provide the `ngrok` URL as the argument to the `callback` parameter when calling the Deepgram API.
2. Check for a POST body size limit on pipedream.com. Some frameworks (e.g. Express) and services restrict incoming request bodies to some size, and that may block Deepgram's callback attempts.

**Reply by rilhia** [reply]
Thanks for the suggestion @jviso. I was being lazy with Pipedream as I didn't want to deal with setting up a local callback server on my machine. The server wouldn't be hard, but dealing with the port forwarding I would need with my router configuration here, would have been too time consuming. I was just trying it out, so took the easy option. But if I were doing this properly, that is exactly the approach I would have taken.

I later checked the POST size limit for Pipedream and found that it is set to 512KB. So this will be the issue. With this in mind, it might be a nice feature to be able to select multiple callbacks based on size limits (for those who do not have access/permission to change it).

**Comment by briancbarrow** [comment]
@rilhia Like @jviso said, that is a good suggestion for an addition to the Usage Logs. I'll bring it up to the team!

**Reply by ujlm** [reply]
Is this feature implemented in the meantime? I need to check the response size belonging to a certain request_id to debug my callback url.
